sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClientBuilder.java

```diff
@@ -146,7 +147,7 @@ public DefaultHttpClientBuilder configuration(Configuration configuration) {
      */
     public HttpClient build() {
         if (httpClientProvider != null) {
-            return httpClientProvider.createInstance();
+            return httpClientProvider.getInstance();
```

Comment:
Isn't this incorrect in general as if things were configured on this builder we lose those configurations?

Reply:
The Javadoc indicates that to be the case. Do we want to change the behavior? Maybe we can make the Javadoc clearer.

Reply:
This builder doesn't have a setter to set the `httpClientProvider`. Where is this set? This seems to always be `null`.

Reply:
I removed this property from both the default and OkHttp builders.